backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/sleketon/BpkSkeleton.kt

```diff
@@ -0,0 +1,189 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clipToBounds
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkColor
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+private val BpkSkeletonBorderRadiusXXS = BpkBorderRadius.Xs.div(2)
+private val BpkHeadlineSkeletonHeightSm = BpkSpacing.Md
+private val BpkHeadlineSkeletonHeightMd = BpkSpacing.Base
+private val BpkHeadlineSkeletonHeightLg = BpkSpacing.Xl
+private val BpkCircleSizeSm = BpkSpacing.Xl
+private val BpkCircleSizeLg = BpkSpacing.Lg.times(2)
+
+enum class BpkSkeletonSizeType {
+    None,
+    Small,
+    Medium,
+    Large
+}
+
+enum class BpkCircleSizeType {
+    None,
+    Small,
+    Large
+}
+
+enum class BpkSkeletonCornerType {
+    Square,
+    Rounded
+}
+private fun Modifier.enhanceHeadlineSize(skeletonSize: BpkSkeletonSizeType) : Modifier {
+    if (skeletonSize === BpkSkeletonSizeType.None) {
+        return this
+    }
+
+    val height = when (skeletonSize) {
+        BpkSkeletonSizeType.Small -> BpkHeadlineSkeletonHeightSm
+        BpkSkeletonSizeType.Medium -> BpkHeadlineSkeletonHeightMd
+        BpkSkeletonSizeType.Large -> BpkHeadlineSkeletonHeightLg
+        BpkSkeletonSizeType.None -> 0.dp
+    }
+
+    return this.height(height)
+}
+
+@Composable
+private fun shimmerBackgroundColor() : Color {
+    return dynamicColorOf(BpkColor.SkyGrayTint06, BpkColor.BlackTint02)
+}
+
+@Composable
+private fun shimmerPrimaryColor() : Color {
+    return dynamicColorOf(Color(0x00FFFFFF), Color(0x00000000))
+}
+
+@Composable
+private fun shimmerSecondaryColor() : Color {
+    return dynamicColorOf(Color(0x99FFFFFF), Color(0x33000000))
+}
+
+@Composable
+private fun shimmerAnimation(): State<Dp> {
+    val infiniteTransition = rememberInfiniteTransition()
+    return infiniteTransition.animateValue(initialValue = (-500).dp, targetValue = 500.dp, typeConverter = Dp.VectorConverter,
+        animationSpec = infiniteRepeatable(animation = tween(durationMillis = 1000, delayMillis = 200, easing = LinearEasing)))
+
+}
+@Composable
+private fun ShimmerBox (modifier: Modifier) {
+    return Box(modifier = modifier
+        .background(
+            brush = Brush.horizontalGradient(
+                listOf(
+                    shimmerPrimaryColor(),
+                    shimmerSecondaryColor(),
+                    shimmerPrimaryColor(),
+                )
+            )
+        )
+    )
+}
+
+@Composable
+fun BpkImageSkeleton (
```

Comment:
looks like there's some formatting issues - there shouldn't be a space here, not sure why this isn't picked up by linters, but could you reformat the file please?

Reply:
ok